runtime/router/trie.go

```diff
@@ -114,14 +121,18 @@ func (t *Trie) Get(path string) (http.Handler, []Param, error) {
 	}
 	// ignore trailing slash
 	path = strings.TrimSuffix(path, "/")
-	return t.root.get(path, false, false, false)
+	colonAsPattern := false
+	if isWhitelisted {
+		colonAsPattern = true
+	}
+	return t.root.get(path, false, false, colonAsPattern, isWhitelisted)
 }
 
 // set sets the handler for given path, creates new child node if necessary
 // lastKeyCharSlash tracks whether the previous key char is a '/', used to decide it is a pattern or not
 // when the current key char is ':'. lastPathCharSlash tracks whether the previous path char is a '/',
 // used to decide it is a pattern or not when the current path char is ':'.
-func (t *tnode) set(path string, value http.Handler, lastKeyCharSlash, lastPathCharSlash bool) error {
+func (t *tnode) set(path string, value http.Handler, lastKeyCharSlash, lastPathCharSlash, colonAsPattern, isWhitelisted bool) error {
```

Comment:
Can you check if this handles URLs like /a/:b/c and /a/:x/d. For whitelist, this router should either give conflict as previous expectation OR if it allows, it should overwrite the other. Also, retrieval of them should be possible using concrete non whitelisted URL. Please add test case for this.

Reply:
There is already similar kind of test case exist. But I can add explicit one.